src/libopensc/reader-pcsc.c

```diff
@@ -1191,13 +1199,6 @@ static void detect_reader_features(sc_reader_t *reader, SCARDHANDLE card_handle)
 		}
 	}
 
-	/* max send/receive sizes: with default values only short APDU supported */
-	reader->max_send_size = priv->gpriv->force_max_send_size ?
-		priv->gpriv->force_max_send_size :
-		SC_READER_SHORT_APDU_MAX_SEND_SIZE;
-	reader->max_recv_size = priv->gpriv->force_max_recv_size ?
-		priv->gpriv->force_max_recv_size :
-		SC_READER_SHORT_APDU_MAX_RECV_SIZE;
```

Comment:
it is not executed if
SCardControl(card_handle, CM_IOCTL_GET_FEATURE_REQUEST, ...
fails, which I have seen on windows systems. So force_max_* is not honored in such cases

Reply:
I see. Then it would be better to put this initialization into `pcsc_add_reader()`, wouldn't it? Otherwise there would be more things that could go wrong before calling `detect_reader_features`...